pkg/reconciler/autoscaling/kpa/resources/decider_test.go

```diff
@@ -161,6 +161,16 @@ func TestMakeDecider(t *testing.T) {
 				d.Spec.InitialScale = 2
 				d.Annotations[autoscaling.InitialScaleAnnotationKey] = "2"
 			}),
+	}, {
+		name: "with min-non-zero-replicas",
+		pa: pa(func(pa *autoscalingv1alpha1.PodAutoscaler) {
+			pa.Annotations[autoscaling.MinNonZeroReplicasKey] = "3"
+		}),
+		want: decider(withTarget(100.0), withPanicThreshold(2.0), withTotal(100),
+			func(d *scaling.Decider) {
+				d.Spec.MinNonZeroReplicas = 3
+				d.Annotations[autoscaling.MinNonZeroReplicasKey] = "3"
```

Comment:
(future cleanup?) is there any reason for the annotation to be propagated if it's on the decider.spec ?

Reply:
Nothing that instantly comes to mind... seems like propagating the annotation is the default behavior (see basically all the cases in this test), so I'll take a look and see if we can clean that up in another PR